### PR TITLE
Sroberge/loading UI

### PR DESCRIPTION
**Pull request by vim-sroberge**

### CODE REVIEW

**Comment by vim-sroberge** [comment]
fix: #76

**Comment by vim-sroberge** [comment]
Fix: #63

**Comment by vim-sroberge** [comment]
Fix: #48

**Comment by vim-sroberge** [comment]
Fix:#23

File: src/ViewerDocument.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    link.href = 'https://vimaec.com'
+    link.appendChild(this.logo)
+    document.body.prepend(link)
+    this.link = link
+
+    ReactDOM.render(<VimUI eventName={stateChangeEventName} />, this.ui)
+  }
+}
+
+function VimUI (props: { eventName: string }) {
```
Comment (cdiggins) [inline src/ViewerDocument.tsx:54]: I believe all of the HTML and CSS for loading and should be set-up in the host HTML page, and passed into the viewer.  I don't think the viewer should be bringing in React. Let's talk.

**Review by mavimaec — COMMENTED** [review]
Adding React as a dependency may be somewhat heavy-handed for this kind of loading UI. It may also impact the way in which other libraries can consume the API. The last time I read about this kind of thing, having different versions of React running on the same page can lead to headaches.

By comparison (and to ease the development of reactive HTML UI), I have previously seen small examples which make use of Vue.js, which can be added to different areas of a web page without impacting the host's dependencies. If you go down the path of Vue.js, make sure you confirm my claims for yourself because my knowledge may also be outdated here.

**Pull request closed without merging**